src/data/data1d/Spectrum1D/ranges/resurrectSpectrum.ts

```diff
@@ -81,6 +91,17 @@ function resurrectSpectrumFromSignals(
       from,
       to,
     });
+
+    let totalIntegration = 0;
+    for (const signal of signals) {
+      const { nbAtoms, atoms = [] } = signal;
+      if ('nbAtoms' in signal) {
+        totalIntegration += Number(nbAtoms);
+      } else if ('atoms' in signal) {
+        totalIntegration += atoms.length;
+      }
```

Comment:
This is a very weird pattern. I would rather do:

```suggestion
      const { nbAtoms, atoms } = signal;
      if (nbAtoms !== undefined) {
        totalIntegration += Number(nbAtoms);
      } else if (atoms !== undefined) {
        totalIntegration += atoms.length;
      }
```